src/core/mem.c

```diff
@@ -224,18 +228,10 @@ static void pp_init(struct page_pool* pool, paddr_t base, size_t size)
         return;
     }
 
-    pages = mem_alloc_ppages(cpu()->as.colors, bitmap_num_pages, false);
-    if (pages.num_pages != bitmap_num_pages) {
-        return;
-    }
-
-    if ((pool->bitmap = (bitmap_t*)mem_alloc_map(&cpu()->as, SEC_HYP_GLOBAL, &pages, INVALID_VA,
-             bitmap_size, PTE_HYP_FLAGS)) == NULL) {
+    if (!mem_bitmap_pool_alloc(bitmap_num_pages, &pool->bitmap)) {
         return;
```

Comment:
Should we return a boolean value in this function so we can check if the allocation fails?

Reply:
Yes. Fixed in 8a4a27cd